### ISSUE TITLE
Find node by ID

### ISSUE THREAD

**Issue by Bones5** [vasturiano/react-force-graph#470]
I am trying to select a node from the graph by its ID and  centre the camera on that nodes position. Is that currently possible?

I feel like I'm missing something but I can't gain access to the graphData once it has been provided to locate the node.

Would a findNodeByyId utility function be possible?

**Comment by vasturiano** [comment]
@Bones5 if you're passing a `graphData` structure to this component then you must have access to that object on your side of the app. 

Once you have the node object you intend to focus the camera, you can simply use `cameraPosition` (in 3D mode) to focus on that particular node's coordinates, a bit like in this example:
https://github.com/vasturiano/react-force-graph/blob/ac46439210387683aae08ceefa29e131c2238d27/example/click-to-focus/index.html#L27-L31

**Comment by Bones5** [comment]
Thanks for the reply.

That is the example I was using, rather than on click though I would like to use an external link/route to navigate to the node.

This means I don't have the position of the node passed from the onClick function, only the ID of the node.

Is there a way to achieve this?

**Comment by vasturiano** [comment]
Yes, you just need to search for the node in your own `graphData` object that you've passed to this component.

**Comment by Bones5** [comment]
Ah, I see now, the position is appended to my original object. 

Sorry, I knew I was missing something.


Thanks very much!
